ml_genn/model.py

```diff
@@ -179,35 +179,35 @@ def evaluate_batched(self, data, time, save_samples=[]):
         spike_t       --  list of spike times for each sample index in save_samples
         """
 
+        n_complete = [0] * len(self.outputs)
         n_correct = [0] * len(self.outputs)
         accuracy = [0] * len(self.outputs)
         all_spikes = [[[] for i,_ in enumerate(self.layers)] for s in save_samples]
 
+        data_remaining = True
         data_iter = iter(data)
         batch_data, batch_labels = next(data_iter)
         batch_i = 0
 
-        batch_labels_queue = deque()
-
-        # Pad number of samples so pipeline can be flushed
-        pipeline_depth = self.calc_pipeline_depth()
-        pipeline_flush = False
-        pipeline_remaining = pipeline_depth
+        # Pipeline queues
+        pipeline_depth = [l.pipeline_depth if hasattr(l, 'pipeline_depth') else 0 for l in self.outputs]
+        pipeline_labels_queue = [deque(maxlen=depth + 1) for depth in pipeline_depth]
 
+        # Process batches
         progress = tqdm()
-        while not pipeline_flush or pipeline_remaining > 0:
+        while data_remaining or any(len(q) > 0 for q in pipeline_labels_queue):
 
-            if not pipeline_flush:
-                batch_data = [np.array(batch_data)]
+            if data_remaining:
+                batch_data = [np.array(d) for d in batch_data]
```

Comment:
I think it would be better to use ``np.asarray`` here as that won't copy if the data is already in a numpy array